libmdnsd/CMakeLists.txt

```
@@ -63,16 +68,12 @@ if (MDNSD_BUILD_FUZZING_CORPUS)
 endif()
 
 if(MDNSD_COMPILE_AS_CXX)
-  set_source_files_properties(${LIBRARY_FILES} PROPERTIES LANGUAGE CXX)
+    set_source_files_properties(${LIBRARY_FILES} PROPERTIES LANGUAGE CXX)
 endif()
 
-if(MDNSD_BUILD_OSS_FUZZ)
-    add_library(libmdnsd ${LIBRARY_FILES} ${PUBLIC_INCLUDES})
-else()
-    add_library(libmdnsd SHARED ${LIBRARY_FILES} ${PUBLIC_INCLUDES})
-endif()
+add_library(mdnsd ${LIBRARY_FILES} ${PUBLIC_INCLUDES})
 
-if(WIN32 OR MINGW)
+if(WIN32)
   target_link_libraries(libmdnsd wsock32 ws2_32)
```

```suggestion
  target_link_libraries(libmdnsd PRIVATE wsock32 ws2_32)
```
Then you can get rid of the `target_link_libraries` in the main cmake script.

---

In the main cmake script, you should change `target_link_libraries(mdnsd libmdnsd)` to
`target_link_libraries(mdnsd PRIVATE mdnsd)`.
The `PRIVATE` will make it use targets.

---

You should check whether this impacts the fuzzer cmake script.
It probably also needs `target_link_libraries(mdnsd PRIVATE mdnsd)`.

---

I still have to use `target_link_libraries(mdnsd PRIVATE libmdnsd)` otherwise CMake complains that the executable mdnsd cannot be linked to another target (itself).

---

But you can remove the wsock2 and ws2_32 to all other executables?